### PR TITLE
add contributing

### PR DESCRIPTION
**Pull request by yzhang123**
Signed-off-by: Yang Zhang <yangzhang@nvidia.com>

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# Pull Requests (PR) Guidelines
+
+**Send your PRs to the `main` branch**
+
+1) Make sure your PR does one thing. Have a clear answer to "What does this PR do?".
+2) Make sure you sign your commits. E.g. use ``git commit -s`` when you commit.
+3) Make sure to add test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
+4) Make sure all unittests finish successfully before sending PR:
+   1) ``pytest`` or (if yor dev box does not have GPU) ``pytest --cpu`` from the root folder
```
Comment (ekmb) [inline CONTRIBUTING.md:14]: TN tests run on CPU only ` @pytest.mark.run_only_on('CPU')`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+5) If you are adding a **new** Python file with a license header, the first line needs to be `Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` (change `2023` to the current year).
+6) If your `text_normalization/LANG/graph_utils.py` is mainly copied from [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
+7) Add ``__init__.py`` for every folder and subfolder.
+8) Remove import guards (`try import: ... except: ...`) if not already done.
+9) follow codeQL results and remove unused variables and imports (report is at the bottom of the PR in github review box)
+10) Add your language support to [tools/text_processing_deployment/export_grammars.sh](tools/text_processing_deployment/export_grammars.sh) and [tools/text_processing_deployment/pynini_export.sh](tools/text_processing_deployment/pynini_export.sh).
```
Comment (jimregan) [inline CONTRIBUTING.md:21]: There's nothing to do for export_grammars.sh; pynini_export.sh should be pynini_export.py

Comment (yzhang123) [reply CONTRIBUTING.md:21]: thanks, let me change it

File: CONTRIBUTING.md (hunk 1 above)
Comment (yzhang123) [reply CONTRIBUTING.md]: thanks added

**Review by ekmb — APPROVED** [review]

**Pull request merged**